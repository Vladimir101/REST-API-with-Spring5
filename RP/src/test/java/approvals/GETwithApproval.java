package approvals;

import org.approvaltests.Approvals;
import org.json.JSONException;
import org.json.JSONObject;
import org.junit.jupiter.api.*;
import org.springframework.http.MediaType;
import org.springframework.test.web.reactive.server.EntityExchangeResult;
import org.springframework.test.web.reactive.server.WebTestClient;

class GETwithApproval
{
	private WebTestClient client;
	private String baseURL = "http://api.zippopotam.us";
	private String endPoint = "/us/94404";
	
	@Test
	void GET() throws JSONException
	{
		client = WebTestClient.bindToServer().baseUrl(baseURL).build();
		
		EntityExchangeResult<String> result = client.get().uri(endPoint)
        	.accept(MediaType.APPLICATION_JSON)
        	.exchange()
        	.expectStatus().isEqualTo(200)
        	.expectHeader().contentType(MediaType.APPLICATION_JSON)
        	.expectBody(String.class)
        	.returnResult();
		
		String body = result.getResponseBody();
		JSONObject json = new JSONObject(body); 
		System.out.println("Pretty print:");
		System.out.println(json.toString(4)); // Print it with specified indentation
		Approvals.verify(json.toString(4));
	}
}
