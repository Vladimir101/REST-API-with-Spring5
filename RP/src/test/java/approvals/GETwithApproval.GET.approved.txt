{
    "country": "United States",
    "places": [
        {
            "latitude": "37.5538",
            "state": "California",
            "state abbreviation": "CA",
            "place name": "San Mateo",
            "longitude": "-122.27"
        }
    ],
    "country abbreviation": "US",
    "post code": "94404"
}